src/Microsoft.AspNet.OData.Shared/DataModificationExceptionType.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.OData.Builder;
+
+namespace Org.OData.Core.V1
+{
+    /// <summary>
+    /// Represents a Message Type
+    /// </summary>
+    public class MessageType
+    {
+        /// <summary>
+        /// Code of message
+        /// </summary>
+        public string Code { get; set; }
+
+        /// <summary>
+        /// Actual message
+        /// </summary>
+        public string Message { get; set; }
+
+        /// <summary>
+        /// Severity of message
+        /// </summary>
+        public string Severity { get; set; }
+
+        /// <summary>
+        /// Target of message
+        /// </summary>
+        public string Target { get; set; }
+
+        /// <summary>
+        /// Details of message
+        /// </summary>
+        public string Details { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception Type
+    /// </summary>    
+    public abstract class ExceptionType
+    {
+        /// <summary>
+        /// Represents a MessageType
+        /// </summary>
+        public MessageType MessageType { get; set; }
+    }
+
+    /// <summary>
+    /// Represents an Exception for Data modification Operation.
+    /// </summary>    
+    public class DataModificationExceptionType : ExceptionType
+    {
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="DataModificationExceptionType"/> class.
+        /// </summary>
+        public DataModificationExceptionType(DataModificationOperationKind failedOperation)
+        {
+            this.FailedOperation = failedOperation;
+        }
+
+        /// <summary>
+        /// Represents king of <see cref="DataModificationOperationKind"/> type of operation 
+        /// </summary>        
+        public DataModificationOperationKind FailedOperation { get; }
+
+        /// <summary>
+        /// Represents response code
+        /// </summary>        
+        public Int16 ResponseCode { get; set; }
```

Comment:
>Int16 [](http://example.com/codeflow?start=15&length=5)

any reason to use Int16?

Reply:
It was taken crom vocabs and its like that so used same.